ledger/src/ui/test/pin/test_pin.c

```diff
@@ -29,48 +29,83 @@
 
 #include "pin.h"
 
+void set_payload(pin_t *pin_ctx, unsigned char *payload, size_t n) {
+    memcpy(pin_ctx->pin_buffer.payload, payload, n);
+}
+
 void test_ok() {
     printf("Test OK...\n");
+    pin_t pin_ctx;
+    reset_pin(&pin_ctx);
 
-    assert(is_pin_valid((unsigned char*)"abcdefgh"));
-    assert(is_pin_valid((unsigned char*)"8b23ef1s"));
-    assert(is_pin_valid((unsigned char*)"MN22P3S9"));
-    assert(is_pin_valid((unsigned char*)"MN22P3S9"));
+    set_payload(&pin_ctx, (unsigned char *)"abcdefgh", strlen("abcdefgh"));
+    assert(is_pin_valid(&pin_ctx));
```

Comment:
Suggestion: replace `set_payload()` + `assert(is_pin_valid())` calls with a call to an `assert_pin` that does both. Then tests could be more readable. Another one for the negative case (or the same with another parameter). You could also use a macro for `true` and `false`, so maybe tests cases could look like:

```
assert_pin("abcdefgh", IS_VALID);
assert_pin("12345678", IS_INVALID);
```

Reply:
That looks great, it will improve readability a lot, I'll change that.